File: app/assets/javascripts/CapabilityStatement.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const CapabilityStatementSchema = DataElementSchema({
  url: Any,
  version: Any,
  name: Any,
  title: Any,
  status: Any,
  experimental: Any,
  date: Any,
  publisher: Any,
  contact: Any,
  description: Any,
  useContext: Any,
  jurisdiction: Any,
  purpose: Any,
  copyright: Any,
  kind: Any,
  instantiates: Any,
  imports: Any,
  software: Any,
  implementation: Any,
  fhirVersion: Any,
  format: Any,
  patchFormat: Any,
  implementationGuide: Any,
  rest: Any,
  messaging: Any,
  document: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'CapabilityStatement' },

});

module.exports.CapabilityStatementSchema = CapabilityStatementSchema;
module.exports.CapabilityStatement = mongoose.model('CapabilityStatement', CapabilityStatementSchema);

File: app/models/qdm/capability_statement.rb
module QDM
  # app/models/qdm/capability_statement.rb
  class CapabilityStatement < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :url
    field :version
    field :name
    field :title
    field :status
    field :experimental
    field :date
    field :publisher
    field :contact
    field :description
    field :useContext
    field :jurisdiction
    field :purpose
    field :copyright
    field :kind
    field :instantiates
    field :imports
    field :software
    field :implementation
    field :fhirVersion
    field :format
    field :patchFormat
    field :implementationGuide
    field :rest
    field :messaging
    field :document
    field :qdmVersion, type: String, default: '4.0.0'
  end
end
